python/dlisio/load.py

```diff
@@ -88,88 +88,209 @@ def load(path, error_handler = None):
+
+    def open_rp66(self):
+        """ Opens RP66 protocol (Visible Envelope part)
+
+        Positions on next VR and opens rp66 protocol from that VR.
+        """
+        core.findvrl(self.stream, self.error_handler)
+        self.stream = core.open_rp66(self.stream)
+
+    def find_offsets(self):
+        """ Finds LR offsets required to parse current file and LF offsets
+        needed to open the next one
+
+        Due to the essence of core.findoffsets function both these things are
+        done at the same time.
+        Code may become more straightforward if findoffsets are ever refactored
+        to deal with this internally and explicitly return the final tell.
+
+        Warning: lfp does *not* make physical tell reliable.
+            We rely on it anyway.
+        """
+        explicits, implicits, broken = core.findoffsets(
+            self.stream, self.error_handler)
+        if len(broken):
+            self.data_eof = True
+
+        self.open_next_at = self.stream.ptell
+        if not self.stream.eof():
+            # rewind only when EOF was detected by reading next LF record
+            # substract VR header size for all the files
+            # substract additional 12 for TIFed files to be positioned before TM
+            self.open_next_at -= 4
+            if self.is_tif:
+                self.open_next_at -= 12
+
+        return explicits, implicits
+
+    def parse_LF(self):
+        """ Parses new logical file
+
+        In the process gathers data about the next logical file
+        """
+        self.open_rp66()
+
+        explicits, implicits = self.find_offsets()
+        recs = core.extract(self.stream, explicits, self.error_handler)
+        sets = core.parse_objects(recs, self.error_handler)
+        pool = core.pool(sets)
+        fdata = core.findfdata(self.stream, implicits, self.error_handler)
+
+        lf = logicalfile(self.stream, pool, fdata, self.sul, self.error_handler)
+        self.lfs.append(lf)
+
+    def eof(self):
+        """ Tests for data EOF
+
+        Returns true only if the physical file reported EOF or data itself is
+        broken such that no new logical file can be opened.
+        """
+        return self.data_eof
+
+    def close(self):
+        """ Closes parser
+
+        Closes current stream along with streams of all processed logical files
+        """
+        self.close_stream()
+        for f in self.lfs:
+            f.close()
+
+    def logical_eof(self):
+        """ Tests for logical EOF
+
+        Decides if currently opened logical stream has data inside.
+        In the process updates parsers knowledge about data eof and next logical
+        file offset.
+        """
+        ltell = self.stream.ltell
+        try:
+            self.stream.get(bytearray(1), ltell, 1)
+        except Exception as e:
+            self.error_handler.log(
+                core.error_severity.critical,
+                "dlis::load: Testing data availability",
```

Comment:
This error message makes zero sense to me. Testing data availbillity? Error on opening new logical file? This function not open a new logical file?